macros/src/codegen.rs

```diff
@@ -355,60 +385,86 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
+                                        } else {
+                                            quote! { self.context.#guard_ident(#temporary_context_call #g_a_ref_param) }
+                                        }
+                                    };
+                                    let guard_expression = expr.to_token_stream(&mut guard_visitor);
+
+                                    if let Some(AsyncIdent { ident: a, is_async: is_a_async }) = action {
+                                        let action_await = match is_a_async {
+                                            true => {
+                                                sm_is_async = true;
+                                                quote! { .await }
+                                            },
+                                            false => quote! { },
+                                        };
+                                        quote! {
+                                            let guard_result = #guard_expression;
+                                            self.context.log_guard(stringify!(#guard_expression), &guard_result);
+                                            match guard_result {
+                                                true => {
+                                                    let _data = self.context.#a(#temporary_context_call #g_a_param) #action_await;
+                                                    self.context.log_action(stringify!(#a));
+                                                    let out_state = #states_type_name::#out_state;
+                                                    self.context.log_state_change(&out_state);
+                                                    self.state = Some(out_state);
+                                                    return self.state()
+                                                },
+                                                false => {},
+                                            }
+                                        }
+                                    } else {
+                                        quote! {
+                                            let guard_result = #guard_expression;
+                                            self.context.log_guard(stringify!(#guard_expression), &guard_result);
+                                            match guard_result {
+                                                true => {
+                                                    let out_state = #states_type_name::#out_state;
+                                                    self.context.log_state_change(&out_state);
+                                                    self.state = Some(out_state);
+                                                    return self.state()
+                                                },
+                                                false => {},
```

Comment:
Replace the match on boolean with simple if statements

Reply:
replaced, thanks